src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/MvcDataAnnotationsMvcOptionsSetup.cs

```diff
@@ -13,33 +13,43 @@ namespace Microsoft.AspNetCore.Mvc.DataAnnotations.Internal
     /// </summary>
     public class MvcDataAnnotationsMvcOptionsSetup : IConfigureOptions<MvcOptions>
     {
-        private IServiceProvider _serviceProvider;
-
-        public MvcDataAnnotationsMvcOptionsSetup(IServiceProvider serviceProvider)
+        private IStringLocalizerFactory _stringLocalizerFactory;
+        private IValidationAttributeAdapterProvider _validationAttributeAdapterProvider;
+        private IOptions<MvcDataAnnotationsLocalizationOptions> _dataAnnotationLocalizationOptions;
+
+        public MvcDataAnnotationsMvcOptionsSetup(
```

Comment:
Could we also add tests for this (with the default ServiceContainer)? This should have failed the setup if `IStringLocalizerFactory` wasn't added to the container.